src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -799,6 +804,765 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetExternalIPv6
     return CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetThreadNetworkDiagnosticAttributeInfo(
+    chip::AttributeId attributeId, uint8_t ** buffer, uint16_t & ReadLength, chip::TLV::TLVType & type)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    switch (attributeId)
+    {
+    case ZCL_CHANNEL_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(uint16_t);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        uint16_t channel = static_cast<uint16_t>(otLinkGetChannel(mOTInst));
+        Encoding::LittleEndian::Put16(*buffer, channel);
+    }
+    break;
+
+    case ZCL_ROUTING_ROLE_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(otDeviceRole);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        otDeviceRole role = otThreadGetDeviceRole(mOTInst);
+        Encoding::LittleEndian::Put16(*buffer, role);
```

Comment:
What guarantees otDeviceRole is not one byte?  Needs a static assert at the very least....

Reply:
This is a mistake and is in fact an enum of one bye. I will correct it

Reply:
Static assert!  ;)